runtime/metricsview/executor_rewrite_percent_of_totals.go

```diff
@@ -0,0 +1,111 @@
+package metricsview
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math"
+	"math/big"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+)
+
+func (e *Executor) rewritePercentOfTotals(ctx context.Context, qry *Query, mv *runtimev1.MetricsViewSpec, sec *runtime.ResolvedSecurity) error {
+	var measures []Measure
+	var measureIndices []int
+	for i, measure := range qry.Measures {
+		if measure.Compute != nil && measure.Compute.PercentOfTotal != nil {
+			measures = append(measures, Measure{
+				Name: measure.Compute.PercentOfTotal.Measure,
+			})
+			measureIndices = append(measureIndices, i)
+		}
+	}
+
+	if len(measures) == 0 {
+		return nil
+	}
+
+	totalsQry := &Query{
+		MetricsView: qry.MetricsView,
+		Measures:    measures,
+		TimeRange:   qry.TimeRange,
+		Where:       qry.Where,
+		TimeZone:    qry.TimeZone,
+	}
+
+	e, err := NewExecutor(ctx, e.rt, e.instanceID, mv, sec, e.priority)
+	if err != nil {
+		return err
+	}
+	defer e.Close()
+
+	res, err := e.Query(ctx, totalsQry, nil)
+	if err != nil {
+		return err
+	}
+	defer res.Close()
```

Comment:
The exactify rewriter does a couple things differently that I think we should also do here:

1. Explicitly set all members of the `Query{}` struct to reduce chance of forgetting to handle one if it's changed in the future
2. Explicitly call `NewAST` and `e.olap.Execute` to a) prevent any chance infinite recursion, b) only apply the rewrites that make sense in this context (I think that would be only `rewriteApproxComparisons` and `rewriteDruidGroups`)

Reply:
Will this not add some complexity? Any new rewrite need to be added in multiple places and the author needs to be aware of it. Not sure if it can be solved as is right now.

Reply:
Yeah it's a little annoying, but I think it's best to explicitly add the rewrites each time at this point. To avoid recursion dangers and because the rewrites are a very mixed bag (some require a specific order of other rewrites, some are expensive to run, some don't make sense in the context of certain calls).

If we end up adding more rewrites, it may make sense to refactor how they're configured to avoid the repetition.